In [79]:
from scipy.stats import uniform, norm
import math

Случайная непрерывная величина A имеет равномерное распределение на промежутке (200, 800].<br> Найдите ее среднее значение и дисперсию.

Непрерывное равномерное распределение (`scipy.stats.uniform`) — непрерывный аналог дискретного равномерного распределения. Функция плотности постоянна внутри отрезка:
$$
    f(x) = \begin{cases}
    1 / (b - a), & x \in [ a, b ], \\
    0 & \text{иначе}.
    \end{cases}
$$

In [17]:
m, d = uniform(loc=200, scale=600).stats()
print( f'Среднее значение: {m}\nДисперсия: {d}')

Среднее значение: 500.0
Дисперсия: 30000.0


Задача 2.  
О случайной непрерывной равномерно распределенной величине B известно, что ее дисперсия равна 0.2.  
Можно ли найти правую границу величины B и ее среднее значение зная, что левая граница равна 0.5?  
Если да, найдите ее.
$$D(X) = \frac{(b-a)^2}{12} = 0.2\,, \quad a = 0.5 \\
b^2 - b - 2.15 = 0 \\
\begin{equation*}
b = 
 \begin{cases}
   -1.04919\\
   2.04919
 \end{cases}
\end{equation*}
$$
Нужный корень лежит справа от левой границы т.е. $b = 2.04919$
   


In [90]:
from sympy import solve, Eq, Symbol
b = Symbol('b')
a, d_x= 0.5, 0.2
b = float(solve(Eq((b-0.5)**2/12, 0.2), b)[1])
m, d = uniform(loc=a, scale= b - a).stats()
print( f'Правая граница b: {b} \nСреднее значение: {m}')

Правая граница b: 2.049193338482967 
Среднее значение: 1.2745966692414834


Задача 3  
Непрерывная случайная величина X распределена нормально и задана плотностью распределения  
$$
   \Large f(x) = \frac{1}{4 \sqrt {2\pi}}e^\frac{-\left({x+2}\right)^{2}}{32}
$$

Найдите:

а). M(X)

б). D(X)

в). std(X) (среднее квадратичное отклонение)

Функция плотности вероятности, совпадающая с функцией Гаусса:
$$
  \Large f(x)={\frac{1}{\sigma {\sqrt {2\pi }}}}e^{-{\frac {1}{2}}\left({\frac {x-\mu }{\sigma }}\right)^{2}}, 
$$ где параметр $\mu$  — математическое ожидание (среднее значение), медиана и мода распределения, а параметр $\sigma$ — среднеквадратическое отклонение, $\sigma ^{2}$ — дисперсия распределения.
Заданная функция с учетом функции плотности вероятности примет вид:
$$
   \Large f(x) = \frac{1}{4 \sqrt {2\pi}}e^{-{\frac {1}{2}}\left(\frac{\left({x - (-2)}\right)}{4}\right)^{2}}
$$
Следовательно, математическое ожидание $M(X) = -2$, среднеквадратическое отклонение $std(X) = 4$, а дисперсия $D(X) = 16$


Задача 4  
Рост взрослого населения города X имеет нормальное распределение.  
Причем, средний рост равен 174 см, а среднее квадратичное отклонение равно 8 см.  
Какова вероятность того, что случайным образом выбранный взрослый человек имеет рост:  
a). больше 182 см  
b). больше 190 см  
c). от 166 см до 190 см  
d). от 166 см до 182 см  
e). от 158 см до 190 см  
f). не выше 150 см или не ниже 190 см  
g). не выше 150 см или не ниже 198 см  
h). ниже 166 см.  

In [89]:
letters = ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h']
lst = [norm.sf(182, loc = 174, scale = 8),
            norm.sf(190, loc = 174, scale = 8),
            norm.cdf(190, loc = 174, scale = 8) - norm.cdf(166, loc = 174, scale = 8),
            norm.cdf(182, loc = 174, scale = 8) - norm.cdf(166, loc = 174, scale = 8),
            norm.cdf(190, loc = 174, scale = 8) - norm.cdf(158, loc = 174, scale = 8),
            norm.cdf(150, loc = 174, scale = 8) + norm.sf(190, loc = 174, scale = 8),
            norm.cdf(150, loc = 174, scale = 8) + norm.sf(198, loc = 174, scale = 8),
            norm.cdf(166, loc = 174, scale = 8)]
for i in range(len(lst)):
    print(f'{letters[i]}: {lst[i]}')


a: 0.15865525393145707
b: 0.022750131948179195
c: 0.8185946141203637
d: 0.6826894921370859
e: 0.9544997361036416
f: 0.024100029979809288
g: 0.0026997960632601866
h: 0.15865525393145707


Задача 5  
На сколько сигм (средних квадратичных отклонений) отклоняется рост человека, равный 190 см, от
математического ожидания роста в популяции, в которой M(X) = 178 см и D(X) = 25 кв.см?


Для вычисления разброса значений нормально распределённой случайной величины можно использовать следующие правила:

* Интервал от $\mu - \sigma$ до $\mu + \sigma$ (__стандартное отклонение__) содержит около $68\%$ вероятностной массы (т.е. с вероятностью $68\%$ данная величина попадает в этот интервал).
* От $\mu - 2\sigma$ до $\mu + 2\sigma$ — около $95\%$ массы (__правило двух сигм__).
* От $\mu - 3\sigma$ до $\mu + 3\sigma$ — около $99.7\%$ массы (__правило трёх сигм__).

Для данной задачи $\mu = 178, \sigma = 5$. Значение x = 190 попадает в интервал заданный __правилом трех сигм__, т.е. $[163, 193]$, а точнее $2,4 \sigma$